### DISCUSSION TITLE
How to add a custom progress loader and how to charge models with Draco compression.

### DISCUSSION THREAD

**Discussion by AYTSistemas13** [google/model-viewer#3922] [Q&A] [answered]
Hi, I have a project in Angular and add the model-viewer to show 3d models, it is confussing to the user notice when the model is ready to interact with it because they don´t put attention to the default progress bar so I don´t know if the tool have already the way to custom the bar or if I can get the event when the model is ready and I just create a custom loader with css. 

Another thing, I compress a .glb file to Draco, because the size was 52mb and I get a file of 5mb, I don´t know if doing this I can cut the time of loading models? and if it is, how can I load this models with a Draco compression?

**Comment by elalish** [comment, marked as answer]
Draco is supported by default, so it should just work. Use our slot to change the progress bar: https://modelviewer.dev/docs/index.html#entrydocs-loading-slots-progressBar - you can find an example in the source of this page: https://modelviewer.dev/examples/lighthouse.html
